components/cdc/src/delegate.rs

```diff
@@ -473,15 +473,20 @@ impl Delegate {
             match entry {
                 Some(KvEntry::RawKvEntry(kv_pair)) => {
                     let mut row = EventRow::default();
-                    decode_rawkv(kv_pair.0, kv_pair.1, &mut row);
-
-                    let row_size = row.key.len() + row.value.len();
-                    if current_rows_size + row_size >= CDC_EVENT_MAX_BYTES {
-                        rows.push(Vec::with_capacity(entries_len));
-                        current_rows_size = 0;
+                    match decode_rawkv(kv_pair.0, kv_pair.1, &mut row) {
+                        Ok(()) => {
+                            let row_size = row.key.len() + row.value.len();
+                            if current_rows_size + row_size >= CDC_EVENT_MAX_BYTES {
+                                rows.push(Vec::with_capacity(entries_len));
+                                current_rows_size = 0;
+                            }
+                            current_rows_size += row_size;
+                            rows.last_mut().unwrap().push(row);
+                        }
+                        Err(e) => {
+                            warn!("cdc decode_rawkv failed"; "error" => ?e);
```

Comment:
Suggest to return error. Warning logs would not be noticed in production environment.